pinot-query-planner/src/test/java/org/apache/pinot/query/validate/PinotTypeCoercionImplTest.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.validate;
+
+import org.apache.pinot.query.QueryEnvironmentTestBase;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class PinotTypeCoercionImplTest extends QueryEnvironmentTestBase {
+
+  @DataProvider(name = "implicitCastCharToIntInWhere")
+  protected Object[][] provideQueriesWithWhere() {
```

Comment:
Please also add examples where two constants are casted. Like `where '10' > 5`. There is where we had issues when trying to remove the coercion from string to binary

Reply:
done